spec/models/role_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Role, type: :model do
+  subject { FactoryGirl.create(:role) }
+
+  describe 'validations' do
+    it { is_expected. to validate_presence_of :name }
+    it { is_expected. to validate_uniqueness_of(:name).case_insensitive }
+    it { is_expected. to be_valid }
+
+    it { is_expected. to have_many :person_roles }
+    it { is_expected. to have_many(:people).through(:person_roles) }
+
+    it 'expects name to be stored lower-case' do
+      subject.name = "EVeNt Planner"
+      subject.save
```

Comment:
This is very close! For readability it might be nicer to write this spec like this:

```ruby
require "rails_helper"

RSpec.describe Role, type: :model do
  it { is_expected.to validate_presence_of :name }
  it { is_expected.to validate_uniqueness_of(:name).case_insensitive }
  it { is_expected.to be_valid }

  it { is_expected.to have_many :person_roles }
  it { is_expected.to have_many(:people).through(:person_roles) }

  context "when the name is mixed-case" do
    let(:role) { FactoryGirl.build(:role, name: "MIxED CaSe") }

    it "saves the name in lower case" do
      role.save!
      expect(role.name).to eq "mixed case"
    end
  end
end
```